rocky/reports/utils.py

```diff
@@ -0,0 +1,27 @@
+import dataclasses
+import logging
+from typing import Dict, List
+
+from django.core.serializers.json import DjangoJSONEncoder
+
+from octopoes.models import OOI
+
+logger = logging.getLogger(__name__)
+
+
+def debug_json_keys(data: Dict, path: List) -> None:
+    for key, value in data.items():
+        if not isinstance(key, (str, int)):
+            logger.error("Key %s is type %s, path is %s", key, type(key), path)
+        if isinstance(value, dict):
+            debug_json_keys(value, path + [key])
+
+
+class JSONEncoder(DjangoJSONEncoder):
+    def default(self, o):
+        if isinstance(o, OOI):
+            return str(o)
+        elif dataclasses.is_dataclass(o):
```

Comment:
can just be an IF

Reply:
The if-elif-else clause here is readable and a common pattern (maybe to be replaced soon by Python's newer structural pattern matching). I don't think it's redundant in this case and I don't have to think about the return strategy of this specific function.